mne_lsl/stream/_filters.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import annotations  # c.f. PEP 563, PEP 649
+
+from copy import deepcopy
+from typing import TYPE_CHECKING
+from warnings import warn
+
+import numpy as np
+from mne.filter import create_filter, estimate_ringing_samples
+from scipy.signal import sosfilt_zi
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    from numpy.typing import NDArray
+
+    from .._typing import ScalarIntType
+
+
+class StreamFilter(dict):
+    """Class defining a filter."""
+
+    def __repr__(self):  # noqa: D105
+        return f"<IIR causal filter ({self['l_freq']}, {self['h_freq']}) Hz>"
```

Comment:
order is usually useful

Reply:
What do you mean by order? Order within `l_freq` and `h_freq` if 2 filters were combined together?

I will update the `repr` anyway for something nicer (even thou `StreamFilter` is part of a private module and will not be exposed publicly, but it might be helpful for renders within a GUI later on).

Reply:
Oh right! Friday evening.. not making the link between the word `order` and the filter order :see_no_evil: 
Do you know/have any code/functions I could use for the for the estimation of the order from the `sos` output?

Reply:
IIRC in `sosfilt` design we always put the first-order section (if present / odd-ordered filter) as the last section. If it's first order, it will have a zero polynomial root. So something like:
```
>>> np.roots(sos[-1, :3])
array([-1.,  0.])
```
tells you the roots, so
```
>>> np.isclose(np.roots(sos[-1, :3]), 0, atol=1e-7).sum()
1
```
being `1` rather than `0` should tell you that it's a first-order filter.